syntax = "proto3";

package connection;
option go_package = "./server/connection";

service ConnectionService {
  rpc Connect(ConnectRequest) returns (ConnectResponse);
  rpc ExecCommand(CommandRequest) returns (CommandResponse);
  rpc PutFile(FileTransferRequest) returns (TransferResponse);
  rpc FetchFile(FileTransferRequest) returns (TransferResponse);
  rpc Close(CloseRequest) returns (CloseResponse);
}

message ConnectRequest {
  string host = 1;
  string user = 2;
  string password = 3;
  int32 port = 4;
}

message ConnectResponse {
  bool success = 1;
  string message = 2;
}

message CommandRequest {
  string command = 1;
}

message CommandResponse {
  int32 exit_code = 1;
  string stdout = 2;
  string stderr = 3;
}

message FileTransferRequest {
  string local_path = 1;
  string remote_path = 2;
}

message TransferResponse {
  bool success = 1;
  string message = 2;
}

message CloseRequest {}

message CloseResponse {
  bool success = 1;
  string message = 2;
}